### ISSUE TITLE
writeShellScriptBin "Is a directory"

### ISSUE THREAD

**Issue by berdario** [NixOS/nixpkgs#30440]
It seems that trivial-builders' `writeShellScriptBin` is not used anywhere in `nixpkgs` and is currently broken

By invoking it like this:

    pkgs.writeShellScriptBin "locale" ''
      echo en_US.UTF-8
    '';

I get an error like:

    building path(s) ‘/nix/store/891wcaj0vwy2ipv34g1kv7clzy9xriwk-locale’
    /nix/store/891wcaj0vwy2ipv34g1kv7clzy9xriwk-locale: /nix/store/891wcaj0vwy2ipv34g1kv7clzy9xriwk-locale: Is a directory
    builder for ‘/nix/store/77mpib3p732hncqdg74051al5b3gxqwi-locale.drv’ failed with exit code 126

I suspect that the following patch might be enough to fix the problem (but I'm a bit in a hurry, and I haven't properly looked into it)

```diff
diff --git a/pkgs/build-support/trivial-builders.nix b/pkgs/build-support/trivial-builders.nix
index 14553c33e0..32214db658 100644
--- a/pkgs/build-support/trivial-builders.nix
+++ b/pkgs/build-support/trivial-builders.nix
@@ -68,7 +68,7 @@ rec {
         ${text}
         '';
       checkPhase = ''
-        ${stdenv.shell} -n $out
+        ${stdenv.shell} -n $out/bin/${name}
       '';
     };
```

**Comment by joachifm** [comment]
Fixed on master by cb593780f3b0d7de4a72346dca208de9df348379

**Comment by bjornfor** [comment]
And NixOS 17.09 (a45cfdba9eab5b0c0f93ad53c41d3e962bdaf6ef).
